src/doc/if-let-guard-review.md

```diff
@@ -0,0 +1,268 @@
+match () {
+    () if true && let 0 = 1 => {}      // error: `let` expressions are unstable (since no feature)
+    () if let 0 = 1 && true => {}      // error: `if let` guards are experimental
+    _ => {}
+}
+```
+
+The test suite checks that these errors mention both the unstable-let and the experimental guard exactly as above.  
+Once the feature is stabilized, these errors will no longer appear.
+
+## Temporaries and Drop Order
+
+A subtle aspect of `if let` guards is the handling of temporaries (and destructor calls) within the guard expression. 
+The Rust reference explains that a *match guard* creates its own temporary scope: any temporaries produced by `GUARD_EXPR` 
+live only until the guard finishes evaluating. Concretely, this means:
+
+* The `guard_expr` is evaluated *after* matching `PAT` but *before* executing the arm’s body (if taken).
+* Any temporary values created during `guard_expr` are dropped immediately after the guard’s scope ends (i.e. before entering the arm body).
+* If the guard fails, those temporaries are dropped right then, and the compiler proceeds to the next arm.
+
+In effect, the drop semantics are the same as for an ordinary match guard or an `if let` in an `if` expression: no unexpected 
+extension of lifetimes. (In Rust 2024 edition, there is a finer rule that even in `if let` 
+expressions temporaries drop before the `else` block; but for match guards the effect is that temporaries from the 
+guard are dropped before the arm body.)
+
+This behavior is exercised by the existing tests. For example, the `drop-order.rs` UI test uses `Drop`-implementing 
+values in nested `if let` guards to verify the precise drop order. Those tests confirm that the values from the inner 
+guards are dropped *first*, before values from outer contexts and before finally moving on to other arms. In short, the 
+feature does not introduce any new irregularity in drop order: guard expressions are evaluated left-to-right 
+(following let-chains semantics) and their temporaries die as soon as the guard completes.
+
+## Lifetimes and Variable Scope
+
+Aside from drop timing, lifetimes of references in the guard work as expected. Because the pattern variables (`PAT` bindings) 
+are in scope during `GUARD_EXPR`, one can take references to them or otherwise use them. Any reference or borrow introduced 
+by the guard is scoped to the guard and arm body. For example:
+
+```rust
+match &vec {
+    v if let [first, ref rest @ ..] = v[..] => {
+        // `first` and `rest` borrowed from `v` are valid here
+        println!("{}", first);
+    }
+    _ => {}
+}
+```
+
+Here `v` is `&Vec`, and the guard borrows parts of it; those references are valid in the arm body. If a guard binds by value 
+(e.g. `if let x = some_moveable`), the usual move/borrow rules apply (see below), but in all cases the scopes follow the match-arm rules.
+
+Moreover, an `if let` guard cannot break exhaustiveness: each arm is either taken or skipped in the usual way. 
```

Comment:
I'd expect exhaustiveness checks to work like with normal guards: they don't play into exhaustiveness checks at all.